server/src/main/java/communication/MessageReceiver.java

```diff
@@ -0,0 +1,16 @@
+package communication;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+
+public class MessageReceiver {
+    public String receiveMessage(BufferedReader reader) {
+        try {
+            return reader.readLine();
```

Comment:
you can create Result string at the begging of the method. It is bad practice to have many return places